python/foglamp/services/common/microservice.py

```diff
@@ -89,3 +91,33 @@ def _get_service_registration_payload(self):
                 "protocol": self._protocol
             }
         return service_registration_payload
+
+    @abstractmethod
+    async def shutdown(self):
+        pass
+
+    @abstractmethod
+    async def change(self):
+        pass
+
+    async def ping(self, request):
+        """ health check
+    
+        """
+        since_started = time.time() - self._start_time
+        return web.json_response({'uptime': since_started})
+
+    async def register(self, request):
```

Comment:
these methods are core only but they must exist (possibly throw error exception instead of passing)